util/eventbus/debughttp.go

```diff
@@ -0,0 +1,200 @@
+package eventbus
+
+import (
+	"embed"
+	"fmt"
+	"html/template"
+	"io"
+	"io/fs"
+	"log"
+	"net/http"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"tailscale.com/tsweb"
+)
+
+type httpDebugger struct {
+	*Debugger
+}
+
+func registerHTTPDebugger(d *Debugger, td *tsweb.DebugHandler) {
+	dh := httpDebugger{d}
+	td.Handle("bus", "Event bus", dh)
+	td.HandleSilent("bus/monitor", http.HandlerFunc(dh.serveMonitor))
+	td.HandleSilent("bus/style.css", serveStatic("style.css"))
+	td.HandleSilent("bus/htmx.min.js", serveStatic("htmx.min.js.gz"))
+	td.HandleSilent("bus/htmx-websocket.min.js", serveStatic("htmx-websocket.min.js.gz"))
+}
+
+//go:embed assets/*.html
+var templatesSrc embed.FS
+
+var templates = sync.OnceValue(func() *template.Template {
+	d, err := fs.Sub(templatesSrc, "assets")
+	if err != nil {
+		panic(fmt.Errorf("getting eventbus debughttp templates subdir: %w", err))
+	}
+	return template.Must(template.ParseFS(d, "*"))
+})
+
+//go:embed assets/*.css assets/*.min.js.gz
```

Comment:
FWIW one pattern I've found useful is to have a `go generate` rule to pin the versions of the library components, e.g.,

```go
// To update the HTMX version, edit the URL here.
//go:generate curl -sL -o static/htmx.min.js https://unpkg.com/htmx.org@v2.0.4/dist/htmx.min.js
```

though if we want them to be compressed we'd probably need to put it in a script and run that. YMMV.

Reply:
Added a go generate program to pull updates.